src/shader_recompiler/ir/passes/resource_tracking_pass.cpp

```diff
@@ -420,32 +420,36 @@ void PatchImageSampleInstruction(IR::Block& block, IR::Inst& inst, Info& info,
                                  Descriptors& descriptors, const IR::Inst* producer,
                                  const u32 image_binding, const AmdGpu::Image& image) {
     // Read sampler sharp. This doesn't exist for IMAGE_LOAD/IMAGE_STORE instructions
-    const u32 sampler_binding = [&] {
+    const auto sampler_binding = [&] -> std::pair<u32, AmdGpu::Sampler> {
         ASSERT(producer->GetOpcode() == IR::Opcode::CompositeConstructU32x2);
         const IR::Value& handle = producer->Arg(1);
         // Inline sampler resource.
         if (handle.IsImmediate()) {
             LOG_WARNING(Render_Vulkan, "Inline sampler detected");
-            return descriptors.Add(SamplerResource{
-                .sharp_idx = std::numeric_limits<u32>::max(),
-                .inline_sampler = AmdGpu::Sampler{.raw0 = handle.U32()},
-            });
+            const auto inline_sampler = AmdGpu::Sampler{.raw0 = handle.U32()};
+            return {descriptors.Add(SamplerResource{
+                        .sharp_idx = std::numeric_limits<u32>::max(),
+                        .inline_sampler = inline_sampler,
+                    }),
+                    inline_sampler};
```

Comment:
```suggestion
            const auto inline_sampler = AmdGpu::Sampler{.raw0 = handle.U32()};
            const u32 sampler_binding = descriptors.Add(SamplerResource{
                .sharp_idx = std::numeric_limits<u32>::max(),
                .inline_sampler = inline_sampler,
            });
            return {sampler_binding, inline_sampler};
```
Small nit but a bit more readable